src_features/provideNFTInformation/cmd_provideNFTInfo.c

```diff
@@ -144,9 +150,11 @@ void handleProvideNFTInformation(uint8_t p1,
     // this prints raw data, so to have a more meaningful print, display
     // the buffer before the endianness swap
     PRINTF("ChainID: %.*H\n", sizeof(chainId), (workBuffer + offset));
-    if ((chainConfig->chainId != 0) && (chainConfig->chainId != chainId)) {
-        PRINTF("Chain ID token mismatch\n");
-        THROW(0x6A80);
+    if (!((chainConfig->chainId == ETHEREUM_MAINNET_CHAINID) &&
+          chain_is_ethereum_compatible(&chainId)) &&
+        (chainId != chainConfig->chainId)) {
```

Comment:
I've read several time and still don't understand the condition, maybe improve or add the clear text equivalent above

Reply:
It is now much simpler. :+1: